### DISCUSSION TITLE
Dynamic Cards in widget

### DISCUSSION THREAD

**Discussion by mikrosmile** [filamentphp/filament#3536] [Help]
I'm having one issue and tried many options, but I think can't solve it myself

Let's say, I have a stats widget which can display Cards, the function returns array of Cards

So, I need to dynamically go through models records and display as many cards as model has, like so:
```
return [
foreach(){
Card.make()
}
```
But I can't pass foreach into array ...

**Comment by danharrin** [comment]
`array_map()`? construct the array and then return it? I'm sorry if this is harsh but it's just PHP

**Reply by mikrosmile** [reply]
Don't know if this is what you suggested, but I figured below code

```
class WorkerOverview extends BaseWidget
{
    protected function getCards(): array
    {
        return collect(Company::all())->map(function ($company) {
            return Card::make($company->name,'123');
        })->toArray();
    }

}
```

Maybe for someone also would be useful. Now in a page I have as many widget Cards as model has there ... :)
